index/index_test.go

```diff
@@ -47,7 +50,9 @@ func TestMicroIndex(t *testing.T) {
 	ctx := context.Background()
 	zctx := zed.NewContext()
 	engine := storage.NewLocalEngine()
-	writer, err := index.NewWriter(ctx, zctx, engine, path, field.DottedList("key"), index.WriterOpts{})
+	writer, err := index.NewWriter(ctx, zctx, engine, path, field.DottedList("key"), index.WriterOpts{
+		ZNGWriterOpts: zngio.WriterOpts{Compress: true},
```

Comment:
Like anyio.WriterOpts am wondering if ZNGWriterOpts should be a pointer and if nil just does the normal thing?

Reply:
My preference would be to remove ZNGWriterOpts since it feels like a knob no one is going to turn.

Reply:
I'm guessing the knob is used/necessary for testing.

Reply:
I certainly agree that it's necessary for the current tests. And though I do think those tests can be made to work without it, that can wait.

What do you say to a separate PR to change index.WriterOpts.ZNGWriterOpts to a pointer?

Reply:
I created #3983 for this.